examples/token/main.aleo

```diff
@@ -17,6 +21,76 @@ record token:
     // The token amount.
     amount as u64.private;
 
+/* Approve */
+
+function approve_public:
+    input r0 as address.public; // approver
+    input r1 as address.public; // spender
+    input r2 as u64.public; // amount spender is allowed to withdraw from approver
+
+    finalize r0 r1 r2;
+
+finalize approve_public:
+    input r0 as address.public; // approver
+    input r1 as address.public; // spender
+    input r2 as u64.public; // increase in amount spender is allowed to withdraw from approver
+
+    hash.ped64 r0 into r3; // hash approver into field
+    hash.ped64 r1 into r4; // hash spender into field
+
+    mul r3 r4 into r5; // multiply approver f by spender f
```

Comment:
Is this a decent way of combining fields into a unique hash? We need the mapping of allowances to be derived from both the owner and spender addresses. Our approach is to hash each one using `hash.ped64` and then multiply the fields together.

Reply:
Something we realized as we were implementing -- this approach won't work in the real world because field multiplication is associative. The addresses 1 & 2 are hashed into fields 1 & 2. Field 1 x Field 2 is equivalent to Field 2 x Field 1. In this case, address 1 approving address 2 would be the same entry as address 2 approving address 1, leading to an easy attack vector for setting allowances for any address.

We still need some way to derive a hash from both addresses in a specified order. We could add Field 1 to the product result, but any guidance on hashing with the addresses would be appreciated.

Reply:
Perhaps you could try defining an `interface` type that consists of two addresses (approver and spender), and then form and hash values of that type. I think that the components of an interface type are serialized in a fixed order before being hashed, and so they shouldn't "commute" as field multiplication does.

Reply:
@acoglio, I've updated the hashing to use the interface type -- it built and the functions executed 👍 Thanks for the suggestion